src/js/angular/guides/tour-lib-services/shepherd.service.js

```diff
@@ -179,15 +207,65 @@ function ShepherdService($location, $translate, LocalStorageAdapter, $route, $in
         return new Shepherd.Tour({
             name: guideId,
             useModalOverlay: true,
-            confirmCancel: true,
-            confirmCancelMessage: $translate.instant('guide.confirm.cancel.message'),
+            confirmCancel: this._confirmGuideCancel,
             defaultStepOptions: {
                 classes: 'shadow-md bg-purple-dark',
                 scrollTo: true
             }
         });
     };
 
+    this._confirmGuideCancel = () => {
+        const guide = Shepherd.activeTour;
+        const currentStep = guide.getCurrentStep()
+
+        if (currentStep.options.isLastStep) {
+            this._completeGuide(guide);
+            return;
+        }
+
+        guide.hide();
+
+        return ModalService.openCustomModal({
+            templateUrl: 'js/angular/guides/templates/modal-confirm-exit.html',
```

Comment:
For some reason when I try to stop the guide via the [x] button I get this error and the guide remains active
![image](https://github.com/user-attachments/assets/18075a38-e44b-4a57-8367-cebb5dc563e1)

Reply:
Could it be that you did not rebuid the app when you switched the workspace?

The tamplate must be coppied by webpack, so a rebuild is needed.